### ISSUE TITLE
4-projects 403 error when applying shared envs

### ISSUE THREAD

**Issue by XSIRIX** [terraform-google-modules/terraform-example-foundation#894]
### TL;DR

When i try to manually apply the shared environments, the strange 403 error with the ADC occurs. Ive checked similar issues in the past regarding this error at this exact step, its all solved tough so i find it very strange that it occured to me.

### Observed behavior

│ Error: Error creating Budget: googleapi: Error 403: Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the billingbudgets.googleapis.com. We recommend configuring the billing/quota_project setting in gcloud or using a service account through the auth/impersonate_service_account setting. For more information about service accounts and how to use them in your application, see https://cloud.google.com/docs/authentication/. If you are getting this error with curl or similar tools, you may need to specify 'X-Goog-User-Project' HTTP header for quota and billing purposes. For more information regarding 'X-Goog-User-Project' header, please check https://cloud.google.com/apis/docs/system-parameters.
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "googleapis.com",
│     "metadata": {
│       "consumer": "projects/764086051850",
│       "service": "billingbudgets.googleapis.com"
│     },
│     "reason": "SERVICE_DISABLED"
│   }
│ ]
│
│   with module.app_infra_cloudbuild_project[0].module.project.module.budget.google_billing_budget.budget[0],
│   on .terraform/modules/app_infra_cloudbuild_project.project/modules/budget/main.tf line 41, in resource "google_billing_budget" "budget":
│   41: resource "google_billing_budget" "budget" {
│
╵

### Terraform Configuration

```hcl
-
```

### Terraform Version

```sh
Terraform v1.3.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.44.1
+ provider registry.terraform.io/hashicorp/google-beta v4.44.1
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

**Comment by bharathkkb** [comment]
Hi @XSIRIX, it looks like a `SERVICE_DISABLED` error. This can happen as service enablement is eventually consistent, so although TF has enabled the API it has not propagated fully. Usually re running after waiting a few mins can fix the issue.

**Comment by jtblin** [comment]
FYI the issue was due to not impersonating the account in my case (because I was encountering errors while doing so)
To fix the impersonation issue I had to give the `Service Account Token Creator` role to the Terraform service account. After that budget creation worked but other things failed later on...
